common/shared.c

```diff
@@ -557,11 +558,19 @@ int ft_getinfo(struct fi_info *hints, struct fi_info **info)
 	if (!hints->ep_attr->type)
 		hints->ep_attr->type = FI_EP_RDM;
 
+	prefix = hints->mode & FI_MSG_PREFIX;
```

Comment:
I don't want to assume 'prefix' semantics based on the hints->mode.  I would rename prefix it to 'force_prefix', and only set it if the -k option has been explicitly requested.

A ```-k=[hint|force]``` option sounds good, but that's more than I was requesting.  I just wanted prefix = 1 only if -k was specified.